From 91bcb2767154cb7b82b71031c1f10f430c6e0524 Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Date: Mon, 25 Mar 2019 10:41:02 +0100
Subject: [PATCH 136/386] crypto: cpt - fix size of scatter/gather list

Number of scatter/gather buffers is 100. This change divides number
of s/g buffers equally between input and output. Previous value of
s/g buffers was too low which caused failures over AF_ALG interface
when page size was set to 4kB.

Change-Id: Ia56ea33c72a6b5681330df77e9e8aa53136cee57
Signed-off-by: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/6422
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Srujana Challa <schalla@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/crypto/cavium/cpt/common/cpt_common.h | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/drivers/crypto/cavium/cpt/common/cpt_common.h b/drivers/crypto/cavium/cpt/common/cpt_common.h
index b68da2a4cc7c..64d16ab36d6c 100644
--- a/drivers/crypto/cavium/cpt/common/cpt_common.h
+++ b/drivers/crypto/cavium/cpt/common/cpt_common.h
@@ -33,9 +33,11 @@
 /* SG list header size in bytes */
 #define SG_LIST_HDR_SIZE	8
 
-/* Maximum number of SG input and output buffers */
-#define MAX_SG_IN_CNT		12
-#define MAX_SG_OUT_CNT		13
+/* Maximum total number of SG buffers is 100, we divide it equally
+ * between input and output
+ */
+#define MAX_SG_IN_CNT		50
+#define MAX_SG_OUT_CNT		50
 
 /* DMA mode direct or SG */
 #define DMA_DIRECT_DIRECT	0
@@ -45,9 +47,6 @@
 #define FROM_CPTR		0
 #define FROM_DPTR		1
 
-/* Maximum number of pending entry input and output buffers */
-#define MAX_BUF_CNT		16
-
 /* CPT instruction queue alignment */
 #define CPT_INST_Q_ALIGNMENT	128
 
@@ -198,8 +197,8 @@ struct cpt_request_info {
 	struct crypto_async_request *areq; /* Async request callback arg */
 	struct cptvf_request req; /* Request information (core specific) */
 	union ctrl_info ctrl; /* User control information */
-	struct buf_ptr in[MAX_BUF_CNT];
-	struct buf_ptr out[MAX_BUF_CNT];
+	struct buf_ptr in[MAX_SG_IN_CNT];
+	struct buf_ptr out[MAX_SG_OUT_CNT];
 	u16 rlen; /* Output length */
 	u8 incnt; /* Number of input buffers */
 	u8 outcnt; /* Number of output buffers */
-- 
2.17.1

